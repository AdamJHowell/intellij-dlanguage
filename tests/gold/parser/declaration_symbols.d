int x,y;   // x and y are ints
int* x,y;  // x and y are pointers to ints
int x,*y;  // error, multiple types
int[] x,y; // x and y are arrays of ints
int x[],y; // error, multiple types