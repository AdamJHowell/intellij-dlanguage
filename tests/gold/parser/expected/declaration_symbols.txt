D Language File
  DLanguageDeclarationImpl(DECLARATION)
    DLanguageDeclaration2Impl(DECLARATION_2)
      DLanguageVariableDeclarationImpl(VARIABLE_DECLARATION)
        DLanguageTypeImpl(TYPE)
          DLanguageType2Impl(TYPE_2)
            DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
              PsiElement(DLanguageTokenType.int)('int')
        DLanguageDeclaratorImpl(DECLARATOR)
          DLanguageIdentifierImpl(IDENTIFIER)
            PsiElement(DLanguageTokenType.ID)('x')
        PsiElement(DLanguageTokenType.,)(',')
        DLanguageDeclaratorImpl(DECLARATOR)
          DLanguageIdentifierImpl(IDENTIFIER)
            PsiElement(DLanguageTokenType.ID)('y')
        PsiElement(DLanguageTokenType.;)(';')
  PsiComment(DLanguageTokenType.line_comment)('// x and y are ints')
  DLanguageDeclarationImpl(DECLARATION)
    DLanguageDeclaration2Impl(DECLARATION_2)
      DLanguageVariableDeclarationImpl(VARIABLE_DECLARATION)
        DLanguageTypeImpl(TYPE)
          DLanguageType2Impl(TYPE_2)
            DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
              PsiElement(DLanguageTokenType.int)('int')
          DLanguageTypeSuffixImpl(TYPE_SUFFIX)
            PsiElement(DLanguageTokenType.*)('*')
        DLanguageDeclaratorImpl(DECLARATOR)
          DLanguageIdentifierImpl(IDENTIFIER)
            PsiElement(DLanguageTokenType.ID)('x')
        PsiElement(DLanguageTokenType.,)(',')
        DLanguageDeclaratorImpl(DECLARATOR)
          DLanguageIdentifierImpl(IDENTIFIER)
            PsiElement(DLanguageTokenType.ID)('y')
        PsiElement(DLanguageTokenType.;)(';')
  PsiComment(DLanguageTokenType.line_comment)('// x and y are pointers to ints')
  PsiElement(DLanguageTokenType.int)('int')
  PsiElement(DLanguageTokenType.ID)('x')
  PsiElement(DLanguageTokenType.,)(',')
  PsiErrorElement:DLanguageTokenType.ID expected, got '*'
    PsiElement(DLanguageTokenType.*)('*')
  PsiElement(DUMMY_BLOCK)
    PsiElement(DLanguageTokenType.ID)('y')
    PsiElement(DLanguageTokenType.;)(';')
    PsiComment(DLanguageTokenType.line_comment)('// error, multiple types')
    PsiElement(DLanguageTokenType.int)('int')
    PsiElement(DLanguageTokenType.[)('[')
    PsiElement(DLanguageTokenType.])(']')
    PsiElement(DLanguageTokenType.ID)('x')
    PsiElement(DLanguageTokenType.,)(',')
    PsiElement(DLanguageTokenType.ID)('y')
    PsiElement(DLanguageTokenType.;)(';')
    PsiComment(DLanguageTokenType.line_comment)('// x and y are arrays of ints')
    PsiElement(DLanguageTokenType.int)('int')
  PsiElement(DLanguageTokenType.ID)('x')
  PsiElement(DLanguageTokenType.[)('[')
  PsiElement(DLanguageTokenType.])(']')
  PsiElement(DLanguageTokenType.,)(',')
  PsiElement(DLanguageTokenType.ID)('y')
  PsiElement(DLanguageTokenType.;)(';')
  PsiComment(DLanguageTokenType.line_comment)('// error, multiple types')